applications/system-service/systemapi.h

```diff
@@ -611,7 +613,7 @@ private slots:
             return;
         }
         auto touch = touches.first();
-        if(swipeDirection != None){
+        if(swipeDirection != None || touch->x == NULL_TOUCH_COORD || touch->y == NULL_TOUCH_COORD){
```

Comment:
I've tried this at first, but this fails (some event are missed).

If the event was complete, then the code would generate a DOWN event.
The next report then would generate a MOVE event.

Typically, what happens is that an incomplete touch is received for slot 0 (for example, missing X).
If you don't ignore it in the SYN_REPORT handler, then a DOWN event is generated (it goes to the pressed list).
Then when the complete event is received, a MOVE event is generated while it should be a DOWN event.
So for small sampling, where you'd get DOWN, MOVE, UP from the driver, it's actually treated as DOWN, MOVE, MOVE, UP, with the first DOWN having a invalid position so it'll never be treated as a swipe with this patch.

Reply:
The result is that when you swipe from the bottom of the screen, it doesn't trigger Oxide's launcher anymore, the log are full of "Swipe cancelled".

Reply:
Gotcha, I'll just need to fabricate the touchDown call when the X or Y was `NULL_TOUCH_COORD` and the next event is a move that fills it in.

Reply:
Didn't my patch worked ?
It's simply patching inside the SYN_REPORT case to ignore it, so everything else works as expected.
It's not dealing with lists so it shouldn't require fixing the malloc stuff size either.

Reply:
While your patch may work, I'd rather use the `modified` and `existing` flags that I implemented the way I intended for them to be used.

Reply:
Okay, I only set `existing` if it's a valid event, which will force touchDown handling instead of skipping the touchMove.